### PR TITLE
FATES API35 parameter file update

### LINKED ISSUE #2478: FATES Sat phenology test failing restart starting in ctsm5.1.dev174
### Brief summary of bug

The test on Derecho

ERS_Ld30.f45_f45_mg37.I2000Clm51FatesSpCruRsGs.derecho_intel.clm-FatesColdSatPhen

fails the restart comparison.

### General bug information

**CTSM version you are using:** Seen in ctsm5.1.dev175, ctsm5.1.dev176 and in what will be ctsm5.2.0

**Does this bug cause significantly incorrect results in the model's science?** Maybe? Restart issues can be a problem in the science...

**Configurations affected:** FATES SP mode.

### Details of bug

@glemieux found it works in ctsm5.1.dev174 but fails in later versions.

### Important details of your setup / configuration so we can reproduce the bug

Interestingly this test

ERP_P128x2_Ld30.f45_f45_mg37.I2000Clm60FatesSpCruRsGs.derecho_intel.clm-FatesColdSatPhen

has been passing. I've verified that there are only two differences:

ERS vs ERP
PE layout of 128x2 vs the default 

The test difference  just means the restart portion is on half the number of processors. So running 64 MPI tasks with one thread.

The PE layout difference is sequential with 2 nodes, threaded verses the default which is a concurrent one with a node for datm, and 4 nodes for CTSM, with no threads.

### Important output or errors that show the problem

The cprnc.out files in TestStatus.log show no differences in the cpl files. But, four fields differ in the CTSM history files as follows:

```
 RMS FATES_MEANLIQVOL_DROUGHTPHEN_PF  3.3974E-02            NORMALIZED  6.8104E-02
 RMS FATES_L2FR_CANOPY_REC_PF         3.0387E-02            NORMALIZED  3.0401E-02
 RMS FATES_L2FR_USTORY_REC_PF         3.0387E-02            NORMALIZED  3.0401E-02
 RMS FATES_MEANLIQVOL_DROUGHTPHEN_PF  3.3974E-02            NORMALIZED  6.8104E-02
```

### PR DESCRIPTION
**Pull request by glemieux**
### Description of changes

This updates the default FATES parameter file and should be coordinated with https://github.com/NGEET/fates/pull/1128.  The fates externals tag updates also brings with it a number of fates scientific updates and bug fixes since tag `sci.1.72.2_api.34.0.0`.  

This PR also cleans up the expected failures list to remove out-of-date cheyenne fates testmods.  It also adds a new test that is expected to fail per #2423.

Finally, it adds build namelist code to make fates hydro and sp mode incompatible for the time being.

### Specific notes

Contributors other than yourself, if any: @rgknox 

CTSM Issues Fixed (include github issue #):
  Fixes #2478

Are answers expected to change (and if so in what way)? Yes, but only for fates.  One new expected failure for `aux_clm` that has been added.

Any User Interface Changes (namelist or namelist defaults changes)? new fates parameter file, check for a failure condition

Testing performed, if any:  standard FATES testing

### CODE REVIEW

**Comment by glemieux** [comment]
To do list on top of the usual list:
- [ ] add fates hydro + sp failure mode unit test
- [ ] test fates hydro + sp failure mode

**Comment by glemieux** [comment]
Running `build-namelist_test.pl` to fail test the new fates namelist graceful failures passed test successfully.

**Comment by glemieux** [comment]
Regression testing is complete and show all expected tests pass B4B on `derecho` with one exception.  `SMS.f10_f10_mg37.I2000Clm50BgcCrop.derecho_nvhpc.clm-cro` failed during the `MODEL_BUILD` step.  All fates diffs are expected due to science updates.

Results: `/glade/u/home/glemieux/scratch/ctsm-tests/tests_pr2436-aux_clm-final`

Regression testing on `izumi` shows all expected tests are B4B.  The only DIFFs are the fates tests which are as expected.

Results: `/home/glemieux/scratch/ctsm-tests/tests_pr2436-aux_clm`

**Review by rgknox — APPROVED** [review]

**Comment by ekluzek** [comment]
@glemieux on the failing test. The clm60 version is marked as failing as well. And it looks like there's a copied version of it. So it probably just needs to have both the clm50 and clm60 version in expected fails. So this looks fine to me. Thanks for pointing it out.

File: cime_config/testdefs/ExpectedTestFails.xml [hunk 1]
```diff
@@ -233,17 +219,17 @@
...
 
-  <test name="PEM_D_Ld15.5x5_amazon.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdSeedDisp">
+  <test name="PEM_D_Ld15.5x5_amazon.I2000Clm50FatesRs.derecho_gnu.clm-FatesColdSeedDisp">
     <phase name="COMPARE_base_modpes">
       <status>FAIL</status>
       <issue>FATES#1089</issue>
     </phase>
   </test>
 
-  <test name="PEM_D_Ld15.5x5_amazon.I2000Clm50FatesRs.derecho_gnu.clm-FatesColdSeedDisp">
```
Comment (ekluzek) [inline cime_config/testdefs/ExpectedTestFails.xml:243]: This actually should be added back in, just for the f10_f10 grid, so I'll do that.

**Review by ekluzek — APPROVED** [review]
Good to bring this in. The CTSM side changes are small, so not a lot to say. There's some small things I noticed in terms of comments for the testlist and Expected fails. Since, @glemieux is out and it's simple I'll do that.

Thanks for doing the testing @glemieux.

**Pull request merged**